### PR TITLE
pkg/release, vendor: add strong types & ctx

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4355.
Towards https://github.com/giantswarm/giantswarm/issues/4337.

### CODE REVIEW

File: pkg/release/version.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package release
+
+type Version string
```
Comment (kopiczko) [inline pkg/release/version.go]: This will allow `:` confusion and prevent us from adding additional function.

File: pkg/release/release.go [hunk 2]
```diff
@@ -120,7 +121,7 @@ func New(config Config) (*Release, error) {
 	return r, nil
 }
 
-func (r *Release) Delete(name string) error {
+func (r *Release) Delete(ctx context.Context, name string) error {
```
Comment (kopiczko) [inline pkg/release/release.go]: I just added `ctx` to every external API call.

File: pkg/release/version.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+func NewStableVersion() Version {
+	return Version(":stable")
```
Comment (MarcelMue) [inline pkg/release/version.go]: This won't be correct here, we do not need the `:` anymore when using helmclient.

File: pkg/release/version.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+func NewVersion(gitSHA string) Version {
+	return Version("@1.0.0-" + gitSHA)
```
Comment (MarcelMue) [inline pkg/release/version.go]: Uncertain if the @ is needed here.

Comment (kopiczko) [reply pkg/release/version.go]: Sho how `helmclient` differentiates between channels and tags?

Comment (kopiczko) [reply pkg/release/version.go]: PullChartTarballFromRelease

Comment (kopiczko) [reply pkg/release/version.go]: Fixed.

File: pkg/release/version.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package release
+
+type Version struct {
+	name      string
+	isChannel bool
+}
```
Comment (kopiczko) [inline pkg/release/version.go]: We can further extend it by adding `NewChannelVersion` and `NewReleaseVersion` but I don't want to expose more than necessary.

**Review by kopiczko — COMMENTED** [review]
@MarcelMue PTALA

**Review by MarcelMue — APPROVED** [review]
We need to try all of this out and see how it works. Change seems fine.

**Pull request merged**